src/high_level/hash.rs

```diff
@@ -67,6 +79,27 @@ pub fn digest(data: &[u8]) -> Result<Digest, UnknownCryptoError> {
     blake2b::Hasher::Blake2b256.digest(data)
 }
 
+/// Hash data from `impl Read` using BLAKE2b-256.
+///
+/// See the [module-level docs](crate::hash) for an example of how to use this function.
+/// Internally calls `std::io::copy()` to move data from the reader into the Blake2b writer.
+/// The `std::io::copy` function buffers reads, so passing in a `BufReader` may be unnecessary.
+///
+/// For lower-level control over reads, writes, buffer sizes, *etc.*, consider using the
+/// [`Blake2b`](crate::hazardous::hash::blake2::blake2b::Blake2b) type and its
+/// [`Write`](std::io::Write) implementation directly. See the Blake2b type's source code
```

Comment:
Again, really great docs. I really like the convenience of linking to `std`-related functionality on the docs, like you've enabled for the `Write`. I was hoping we could also do this for the mentions of:

- `Read`
- `std::io::copy()` (since this already a full path it doesn't even need a link, just [])
- `BufReader`

Lastly, maybe "See the Blake2b type's source code [..]" could be "See the Blake2b Write implementation source code [..]" for a small bit of added clarity on where to go look. Or point to the test if that is what is meant.

Reply:
All great points. I'll make the changes.